pandas/core/internals/managers.py

```diff
@@ -711,16 +711,16 @@ def combine(self, blocks, copy=True):
         return type(self)(new_blocks, axes, do_integrity_check=False)
 
     def get_slice(self, slobj: slice, axis: int = 0):
-        if axis >= self.ndim:
-            raise IndexError("Requested axis not found in manager")
 
         if axis == 0:
             new_blocks = self._slice_take_blocks_ax0(slobj)
-        else:
+        elif axis == 1:
             _slicer = [slice(None)] * (axis + 1)
             _slicer[axis] = slobj
             slicer = tuple(_slicer)
             new_blocks = [blk.getitem_block(slicer) for blk in self.blocks]
+        else:
+            raise IndexError("Requested axis not found in manager")
```

Comment:
self.ndim==1 and axis==1 are guarded against elsewhere?

Reply:
yes